AdobeStockImage/Model/SaveImage.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\AdobeStockImage\Model;
+
+use Magento\AdobeStockImageApi\Api\SaveImageInterface;
+use Magento\Framework\Exception\CouldNotSaveException;
+use Magento\AdobeStockClientApi\Api\ClientInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Class SaveImage
+ */
+class SaveImage implements SaveImageInterface
```

Comment:
Looks like this service is applicable only to licensed images. I would indicate that in the naming. However, if it will accept URL instead of adobeId - this service can be applicable for both licensed and preview images